app/src/main/java/com/example/pokemonapp/fragment/SplashScreenFragment.kt

```diff
@@ -0,0 +1,41 @@
+package com.example.pokemonapp.fragment
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
+import androidx.navigation.findNavController
+import com.example.pokemonapp.databinding.FragmentSplashScreenBinding
+import com.example.pokemonapp.utils.Event
+import com.example.pokemonapp.viewmodel.SplashScreenData
+import com.example.pokemonapp.viewmodel.SplashScreenStatus
+import com.example.pokemonapp.viewmodel.SplashScreenViewModel
+import org.koin.androidx.viewmodel.ext.android.viewModel
+
+class SplashScreenFragment : Fragment() {
+
+    private lateinit var binding: FragmentSplashScreenBinding
+    private val viewModel by viewModel<SplashScreenViewModel>()
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+        binding = FragmentSplashScreenBinding.inflate(inflater, container, false)
+        viewModel.getLiveDataState().observe(viewLifecycleOwner, updateUIObserver)
+        viewModel.navigateToMainFragment()
+        return binding.root
+    }
+
+    private val updateUIObserver = Observer<Event<SplashScreenData>> { event ->
+        val eventContent = event.getContentIfNotHandled()
```

Comment:
you can avoid this variable, and just call it on when

Reply:
done